[WIP] Uncomment docs of tango.Util.instance()

reszelaz opened the pull request:
Hi,

Just in case it helps others, I thouht it may be useful to add to the documentation the `tango.Util.instance()` argument as it was explained in [this forum post](https://www.tango-controls.org/community/forum/c/development/python/how-to-check-if-the-code-is-being-executed-by-a-server-process/?page=1#post-4511).

I saw that it was already documented and I just tried to uncomment it. However it still does not generate the docs. Well, I just made some quick tries, but if you think it is interesting to add this to the docs, I could investigate further.

Many thanks!

reszelaz committed:
Uncomment docs of tango.Util.instance()

ajoubertza commented:
Thanks, @reszelaz.  Improved documentation is always welcome, so please see if you can find a solution.  If the method cannot be easily documented that way, then we could add a note at the end of the general class docstring, `Util.__doc__`, instead.

ajoubertza commented on tango/pyutil.py:
```diff
@@ -296,21 +296,21 @@ class and its constructor is not public. Example::
+
+           Static method that gets the singleton object reference.
+           If the class has not been initialised with it's init method,
+           this method prints a message and aborts the device server process
+
+       Parameters :
+           - exit : (bool)
```
Would be useful to add more detail somewhere here.  E.g., it raises a DevFailed exception instead of aborting if `exit` is set to `False`.

reszelaz committed:
Add more details about exit parameter and throwing DevFailed

reszelaz committed:
Patch tango.utils.document_static_method in mock Sphinx extension

As of today, this enables documentation for the following static methods:
- ApiUtil: instance() and cleanup()
- Connection: get_fqdn()
- Exception: throw_exception(), re_throw_exception(), print_error_stack(), print_exception(), throw_python_exception(),
- Util.instance()
